app/src/main/java/org/oppia/app/drawer/NavigationDrawerFragmentPresenter.kt

```diff
@@ -74,6 +75,16 @@ class NavigationDrawerFragmentPresenter @Inject constructor(
   private fun subscribeToProfileLiveData() {
     getProfileData().observe(fragment, Observer<Profile> {
       navigationDrawerHeaderViewModel.profileName.set(it.name)
+      when {
+        it.isAdmin -> {
+          binding.administratorControlsLinearLayout.visibility = View.VISIBLE;
```

Comment:
Try contrtolling this using ViewModel instead directly controlling visibility hear.

Reply:
Done.